packages/sources/coinmarketcap-test/src/endpoint/crypto.ts

```diff
@@ -1,25 +1,270 @@
+  crypto: 'price',
+  volume: 'volume_24h',
+  marketcap: 'market_cap',
+} as const
 
 const httpTransport = new HttpTransport<CryptoEndpointTypes>({
-  prepareRequests: (params, config) => {
-    return buildBatchedRequestBody(params, config)
+  prepareRequests: (params: CryptoRequestParams[], settings) => {
+    const requests = []
+    const groupedParams = {
+      id: [],
+      slug: [],
+      symbol: [],
+    } as Record<string, CryptoRequestParams[]>
+
+    for (const param of params) {
+      if (param.cid) {
+        groupedParams.id.push(param)
+      } else if (param.slug) {
+        groupedParams.slug.push(param)
+      } else if (param.base) {
+        groupedParams.symbol.push(param)
+      } else {
+        logger.error(
+          `Params were not able to be classified into ID, Slug or Symbol: (${JSON.stringify(
+            param,
+          )})`,
+        )
+      }
+    }
+
+    for (const [idType, fullList] of Object.entries(groupedParams)) {
+      // CMC does not support more than 120 unique quotes
+      const chunkedList = chunkArray(fullList, 120)
+
+      // This could be further optimized in cases with more than 120 entries, to make sure that
+      // chunkes are grouped optimally to avoid sending unnecessary converts
+      for (const list of chunkedList) {
+        requests.push({
+          params: list,
+          request: {
+            baseURL: settings.API_ENDPOINT,
+            url: '/cryptocurrency/quotes/latest',
+            headers: {
+              'X-CMC_PRO_API_KEY': settings.API_KEY,
+            },
+            params: {
+              [idType]: [...new Set(list.map((p) => p.cid || p.slug || p.base))].join(','),
+              convert: [...new Set(list.map((p) => p.quote))].join(','),
+            },
+          },
+        })
+      }
+    }
+
+    return requests
   },
   parseResponse: (params, res) => {
-    return constructEntry(params, res.data, 'price')
+    logger.debug(`CMC api call cost: ${res.data.cost}`)
+
+    // The params are already grouped by ID type, so we only have to look at the first one
+    const idType = params[0].cid ? 'id' : params[0].slug ? 'slug' : 'symbol'
+
+    // Build a mapping to ids
+    const idsMap: Record<string, string> = {}
+    for (const data of Object.values(res.data.data)) {
+      idsMap[data[idType]] = data.id.toString()
+    }
+
+    // Use the mapping to generate the responses
+    return params.map((p) => {
+      const id = idsMap[p.cid || p.slug || p.base]
+      if (!id) {
+        return {
+          params: p,
+          response: {
+            statusCode: 502,
+            errorMessage: `An ID was not found in the idsMap for parameter ${JSON.stringify(p)}`,
+          },
+        }
+      }
+
+      const data = res.data.data[id]
+      if (!data) {
+        return {
+          params: p,
+          response: {
+            statusCode: 502,
+            errorMessage: `ID type "${idType}" was not found in response for request: ${JSON.stringify(
```

Comment:
If I'm right in the last comment, we might need to update this too
```suggestion
            errorMessage: `Data was not found in response for request: ${JSON.stringify(
```